### ISSUE TITLE
License issue, continued

### ISSUE THREAD

**Issue by cjvanlissa** [aaronpeikert/repro#24]
I remember now why I wanted to use GPL over MIT: because it prevents others from making a for-profit "copy of" worcs, and because I want to encourage others to share their code.

I think we should at least consider adopting the same license; here's a nice page discussing the benefits and disadvantages of each https://exygy.com/blog/which-license-should-i-use-mit-vs-apache-vs-gpl/

Collaborators Barbara Vreede and Anna-Lena Lamprecht are also quite knowledgeable in this area.

**Comment by aaronpeikert** [comment]
cross reference https://github.com/cjvanlissa/worcs/issues/23

**Comment by aaronpeikert** [comment]
Yes, I think we should consider it. However, as far as I understand, a central problem is that the legal definition of "commercial" is rather narrow and includes stuff like teaching at for-profit universities and research within third-party funds. At least it is my understanding that there is considerable debate.

**Comment by aaronpeikert** [comment]
I could e.g. double license repro as MIT + GPL3 or we could reach out to Barbara Vreede and Anna-Lena Lamprecht or one of the lawyers at the MPIB?

**Comment by cjvanlissa** [comment]
As far as I understand it, those people would only be prohibited from making derivative works, not from using the program

**Comment by aaronpeikert** [comment]
Since R itself is GPL and most of the packages repro is relying on as well, we can move to GPL if you are more comfortable with it. I am not a fan of the copyleft idea, but as almost everything in R has it; I am overruled :neutral_face:

**Comment by cjvanlissa** [comment]
Alright, I think that's good. We'll get some input from those other people, and if nobody object we can all use GPL

**Comment by aaronpeikert** [comment]
I now remember what that my argumentations above (with commercial use) doesn't relate at all to GPL vs MIT but CC0 vs CC-BY.
As I made the decision on my own to use MIT I take the liberty to switch to GPL.

**Comment by aaronpeikert** [comment]
However, maybe we should discuss the licensing of the non-code parts (namely all sorts of documentation). I would suggest we use CC0 for everything (function documentation and vignettes) and rely on good scientific practice for citation (instead of forcing people by use of CC-BY).

What are your thoughts @cjvanlissa? Have you already thought about which license to use for your paper?

I am thinking about including a scientific paper or some version of it as a vignette in order to have an always up to date version out there if things with the software change.

**Comment by cjvanlissa** [comment]
The choice of license depends in part on the journal. There is usually a tradeoff between journal reputation and license strictness. Andreas had suggested AMPPS, and I had considered psych methods.

**Comment by aaronpeikert** [comment]
I know, I know, but it is maybe its worth to ask ([even nature uses CC0](https://www.nature.com/nature-research/editorial-policies/self-archiving-and-license-to-publish)).

**Comment by cjvanlissa** [comment]
Basically ... I lack the expertise to decide this. I feel a bit apprehensive about using different licenses on the package code and package documentation (I've also never seen this). And the paper license will be determined by the outlet. I'll ask around.

**Comment by aaronpeikert** [comment]
GPL authors do recommend licensing the documentation differently than the code (see [FAQ](https://www.gnu.org/licenses/gpl-faq.html#WhyNotGPLForManuals)).
But maybe it is only fruitful for long-form documentation and other educational material we create.

**Comment by bvreede** [comment]
I have taken this issue as a reason to dive into GPL and *gulp* actually _read_ the license.

First off: wow, that was an interesting and challenging exercise, and I'm very convinced it is worth it to do some sort of... journal club for software licenses. No joke.

Second: after reading, I now realize that I have misunderstood GPL and I see that same misunderstanding reflected above. GPL is not a non-commercial license, and it fully allows for-profit use of software, documentation, assistance, and associated activities. Their objective is for GPL projects to be open source, but not necessarily free-of-charge.

The confusion here, and the reason some people/companies shy away from using it, is that it contains a fairly strict share-alike policy. Meaning: if you use GPL, and someone else wants to use your software in their product, they need to license this under GPL as well. And so forth.

It is not fully clear to me yet if this pertains to use of the language or even a library as well — in that case, everything written in R should be licensed under GPL, and the same for everything using tidyverse libraries. While this would appear illogical to me, there is a clause in GPL about using software as an object, which... is that what it means? I'm not sure. Hence my desire to get some bright heads together on this issue.

In conclusion: I would say that GPL is a wise choice and makes sense for worcs as well as repro. It should not prohibit commercial use of the product (though this can probably be added with a custom addendum to the license if that is a strong desire).

With regard to the use of different licenses: this does not seem all that odd to me, given that the nature of software and its documentation is very different. That said, it will be different to do this when you're talking about a vignette (tightly linked to the software & usually part of an R package) versus a paper (published in a different venue altogether).

Regarding citation, GPL does not specify that attribution is given when the software is used, only that any modified version of it be open source and GPL-licensed. Personally I would feel comfortable using CC-BY, and not consider it "forcing citation", especially since these packages are for use in scientific circles in which citation practices are common anyway, but this may all be a moot point as a journal may offer a limited selection of licenses.

**Comment by cjvanlissa** [comment]
@aaronpeikert @bvreede and this is why we're lucky to have Barbara involved :)

Thank you so much, Barbara! I agree with your assessment, and would be inclined to stick with GPL for worcs at least. Aaron, the ball is in your court!

**Comment by aaronpeikert** [comment]
Thank you @bvreede for struggling through this for us!
Regarding licensing documentation/vignette with CC0 or CC-BY:
I think this is important because I expect us to invest a lot of resources and time into "educational" material and subsequently want users to feel comfortable using these materials. Personally I only feel really at ease if I see a CC license.
As the vignette is distributed with the package you are correct that the license of the package also applies to the vignette. However, there is no rule to stop us from "dual-licensing" and declaring that a vignette is also under CC0. Technically this allows the license taker to choose between GPL and CC0.

Way ahead of you @cjvanlissa: aaronpeikert/repro@b649f2a38b1e77a7e11a02bb28fad6b7800ab80e is five days old :partying_face:
